TeamCode/src/main/java/org/firstinspires/ftc/teamcode/ViperArrrrrm.java

```diff
@@ -19,17 +21,16 @@ public class ViperArrrrrm extends LinearOpMode {
 
     void initialize(HardwareMap hwm){
         viperArm = hwm.get(DcMotor.class, Constants.VIPER_ARM);
-        viperArm.setDirection(DcMotor.Direction.FORWARD);
+        viperArm.setDirection(DcMotor.Direction.REVERSE);
     }
-    void operateArm(double armPower){
+    void operateArm(Telemetry telemetry, double armPower){
         double max = 1;
         max = Math.max(max, Math.abs(armPower));
         if (armPower > max) {
             armPower = max;
         }
         viperArm.setPower(armPower);
         telemetry.addData("arm power", "%4.2f", armPower);
-        telemetry.update();
     }
```

Comment:
We have an encoder on this motor, as well as a limit switch on the bottom to zero out the encoder. 
Examples:
* Touch Sensor: https://github.com/RedArrowRobotics/2024-21353-RobotCode/blob/master/FtcRobotController/src/main/java/org/firstinspires/ftc/robotcontroller/external/samples/SensorTouch.java
* Encoder (The example is using drive motors, so there will be some differences in how we want to control it, but this should get you started): https://github.com/RedArrowRobotics/2024-21353-RobotCode/blob/master/FtcRobotController/src/main/java/org/firstinspires/ftc/robotcontroller/external/samples/RobotAutoDriveByEncoder_Linear.java

Reply:
essentially - we will want to zero out the encoder whenever the button is pressed, then add some telemetry so we can figure out what encoder counts correspond to heights of the two buckets. Then use 'runToPosition' to go directly to either the upper or lower basket.